parser.py

```diff
@@ -1,17 +1,49 @@
 #!/bin/env python
 import argparse
 import pprint
+import re
+from abc import abstractmethod
+from typing import override
+
 import grammar
 import visitor
 
 
+class Preprocessor:
+    @abstractmethod
+    def process(self, program: str):
+        pass
+
+
+class CommentsPreprocessor(Preprocessor):
+
+    @override
+    def process(self, program: str):
+        cleaned_lines = filter(
+            lambda cleaned_line: cleaned_line,
```

Comment:
What's this lambda doing...?

Reply:
To remove the empty lines that were giving me problems (an empty string gives false). It is now solved with the change of the program grammar